Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py

```diff
@@ -47,5 +46,25 @@ def test_default_options_d2b(self):
         self.assertTrue(item.getRun())
         self.assertTrue(item.getRun().hasProperty('run_number'))
 
+    def test_default_options_d20(self):
```

Comment:
So I'm happy with all the changes here however hesitant on this test falling afoul of our [unit test good practice](https://developer.mantidproject.org/UnitTestGoodPractice.html#other-more-general-points) and our unit tests slowing down by lots of little regressions like this (akin to "death by a thousand paper cuts")

Prior to this change the PowderILLDetectorScanTest takes 6s on [isis-ndw1297](https://builds.mantidproject.org/job/pull_requests-ubuntu/39762/consoleFull) (which isn't great) but adding this takes an additional 2 seconds on the [same machine](https://builds.mantidproject.org/job/pull_requests-ubuntu/39675/consoleFull).

Could we either par down 171414 to be a representative subset of the data (say using 10% of the original spectra) or move it to a system test?

Reply:
It depends, it happens that if you reduce/mock the data the test passes, but it fails with the real data. I'm not saying this is the case here, but it is always a tradeoff between skinning your data or keeping it real. In general, the strategy was to have a unit test that covers the default options, this is what is added, and then have more complete scenarios in the system test. There are many other unit test suits that run over 10s, so for me up 10s have always been acceptable. We can of course discuss later if we want to introduce stricter criteria project-wide, but I don't think it should block this PR.

Reply:
I agree that it's not fair to block this PR if you have already tried reducing / mocking the data, typically when I bring this up people haven't realised the impact of putting a full dataset in.

On a local developer machine it's less apparent, but you tend to find when a release is looming that one CI job that randomly failed and now will take another 2 hours to run unit tests tends to stir up annoyance. But equally that's something for developers as a whole to raise if it becomes a problem.